usermods/sht/usermod_sht.h

```diff
@@ -0,0 +1,312 @@
+      *    Instead, use a timer check as shown here.
+      */
+    void loop()
+    {
+      if (!enabled || !initDone || strip.isUpdating()) return;
+
+      if (isShtReady()) {
+        if (millis() - shtLastTimeUpdated > 30000 && !shtDataRequested) {
+          shtTempHumidSensor->requestData();
+          shtDataRequested = true;
+
+          shtLastTimeUpdated = millis();
+        }
+
+        if (shtDataRequested) {
+          if (shtTempHumidSensor->dataReady()) {
+            if (shtTempHumidSensor->readData(false)) {
+              shtCurrentTemp = shtTempHumidSensor->getTemperature();
+              shtCurrentHumidity = shtTempHumidSensor->getHumidity();
+
+              publishTemperatureAndHumidityViaMqtt();
+              shtReadDataSuccess = true;
+            }
+            else {
+              shtReadDataSuccess = false;
+            }
+
+            shtDataRequested = false;
+          }
+        }
+      }
+    }
+
+    void onMqttConnect(bool sessionPresent) {
+      if (haMqttDiscovery) publishHomeAssistantAutodiscovery();
+    }
+
+    void addToConfig(JsonObject &root)
+    {
+      JsonObject top = root.createNestedObject(FPSTR(_name)); // usermodname
+
+      top[FPSTR(_enabled)] = enabled;
+      top[FPSTR(_haMqttDiscovery)] = haMqttDiscovery;
+    }
+
+    /**
+     * readFromConfig() is called before setup() to populate properties from values stored in cfg.json
+     *
+     * The function should return true if configuration was successfully loaded or false if there was no configuration.
+     */
+    bool readFromConfig(JsonObject &root)
+    {
+      JsonObject top = root[FPSTR(_name)];
+      if (top.isNull()) {
+        DEBUG_PRINTF("[%s] No config found. (Using defaults.)\n", _name);
+        return false;
+      }
+
+      bool oldEnabled = enabled;
+      bool oldHaMqttDiscovery = haMqttDiscovery;
+      
+      getJsonValue(top[FPSTR(_enabled)], enabled);
+      getJsonValue(top[FPSTR(_haMqttDiscovery)], haMqttDiscovery);
+
+      // First run: reading from cfg.json, nothing to do here, will be all done in setup()
+      if (!firstRunDone) {
+        DEBUG_PRINTF("[%s] First run, nothing to do\n", _name);
+      }
+      // Check if mod has been en-/disabled
+      else if (enabled != oldEnabled) {
+        enabled ? setup() : cleanup();
+        DEBUG_PRINTF("[%s] Usermod has been en-/disabled\n", _name);
+      }
+      // Config has been changed, so adopt to changes
+      else if (enabled) {
+        if (oldHaMqttDiscovery != haMqttDiscovery && haMqttDiscovery) {
+          publishHomeAssistantAutodiscovery();
+        }
+
+        DEBUG_PRINTF("[%s] Config (re)loaded\n", _name);
+      }
+
+      return true;
+    }
+
+    void addToJsonInfo(JsonObject& root)
+    {
+      if (!enabled && !isShtReady()) {
+        return;
+      }
+
+      JsonObject user = root["u"];
+      if (user.isNull()) user = root.createNestedObject("u");
```

Comment:
For some strings use F() helper to store them in flash memory (not needed for "u" but others may benefit).